src/main/java/fi/aalto/cs/apluscourses/intellij/utils/CourseFileManager.java

```diff
@@ -192,9 +203,18 @@ private void loadFromJsonObject(@NotNull JSONObject jsonObject) throws IOExcepti
     Iterable<String> moduleNames = modulesObject::keys;
     for (String moduleName : moduleNames) {
       JSONObject moduleObject = modulesObject.getJSONObject(moduleName);
+
       String moduleId = moduleObject.getString(MODULE_ID_KEY);
-      ZonedDateTime downloadedAt
-          = ZonedDateTime.parse(moduleObject.getString(MODULE_DOWNLOADED_AT_KEY));
+
+      String downloadedAtStr = moduleObject.optString(MODULE_DOWNLOADED_AT_KEY);
+      ZonedDateTime downloadedAt;
+      if (downloadedAtStr == null || downloadedAtStr.trim().isEmpty()) {
+        logger.error("Module " + moduleName + " missing 'downloadedAt' in course file");
+        downloadedAt = Instant.EPOCH.atZone(ZoneId.systemDefault());
```

Comment:
Would it be safer to default to a distant day in the past? Because we use this information to test whether or not files have been changed since the module was downloaded, false positives (asking user if they want to overwrite even though there are no changes) cause less harm than false negatives (not asking user when there actually are changes).

Reply:
`Instant.EPOCH` is a constant for `1970-01-01T00:00:00Z`, so it's pretty distant in the past :smile:

Reply:
Oh sorry, you're right. 🤦‍♂️  My mind only read "Instant" and automatically thought it means like "instantly now". I was almost to suggest the epoch day to be chosen as default... 😛